src/wpantund/NCPInstanceBase.h

```diff
@@ -136,38 +136,71 @@ class NCPInstanceBase : public NCPInstance, public EventHandler{
 
 public:
 	// ========================================================================
-	// MARK: On Mesh Prefix Management
+	// MARK: Global address/prefix/route management
 
-	void add_prefix(const struct in6_addr &address, uint32_t valid_lifetime = UINT32_MAX, uint32_t preferred_lifetime = UINT32_MAX, uint8_t flags = 0);
+	enum Origin {
+		kOriginThreadNCP,
+		kOriginPrimaryInterface,
+		kOriginUser,
+	};
 
-public:
-	// ========================================================================
-	// MARK: Global Address Management
+	void unicast_address_was_added(Origin origin, const struct in6_addr &address,
+			uint8_t prefix_len = 64, uint32_t valid_lifetime = UINT32_MAX, uint32_t preferred_lifetime = UINT32_MAX);
 
+	void unicast_address_was_removed(Origin origin, const struct in6_addr &address);
 
-	void add_address(const struct in6_addr &address, uint8_t prefix = 64, uint32_t valid_lifetime = UINT32_MAX, uint32_t preferred_lifetime = UINT32_MAX);
-	void remove_address(const struct in6_addr &address);
+	void multicast_address_was_joined(Origin origin, const struct in6_addr &address);
```

Comment:
nitpick: I normally think of joining "multicast groups", not joining "multicast addresses".